pandas/core/arrays/numpy_.py

```diff
@@ -377,19 +387,28 @@ def arithmetic_method(self, other):
                 if isinstance(a, np.ndarray):
                     # for e.g. op vs TimedeltaArray, we may already
                     #  have an ExtensionArray, in which case we do not wrap
-                    return cls(a), cls(b)
+                    return self._wrap_ndarray_result(a), self._wrap_ndarray_result(b)
                 return a, b
 
             if isinstance(result, np.ndarray):
                 # for e.g. multiplication vs TimedeltaArray, we may already
                 #  have an ExtensionArray, in which case we do not wrap
-                return cls(result)
+                return self._wrap_ndarray_result(result)
             return result
 
         return compat.set_function_name(arithmetic_method, f"__{op.__name__}__", cls)
 
     _create_comparison_method = _create_arithmetic_method
 
+    def _wrap_ndarray_result(self, result: np.ndarray):
```

Comment:
isn't this the same code you added in arrays/integer?

Reply:
similar, but i dont see a nice way to share it in the short term.  medium-term ive got some thoughts kicking around